File: backend/internal/domain/entities/system_stat.go
package entities

import "fmt"

type SystemStat struct {
	HostStat           *HostStat         `json:"host_stat"`
	UsersStat          []*UserStat       `json:"users_stat"`
	MemoryStat         *MemoryStat       `json:"memory_stat"`
	CPUStat            *CPUStat          `json:"cpu_stat"`
	DiskStat           *DiskStat         `json:"disk_stat"`
	PartitionStat      []*PartitionStat  `json:"partition_stat"`
	NetConnectionsStat []*ConnectionStat `json:"net_connections_stat"`
	ProcessesStat      []*ProcessStat    `json:"processes_stat"`
}

func NewSystemStat(
	hostStat *HostStat,
	usersStat []*UserStat,
	memoryStat *MemoryStat,
	cpuStat *CPUStat,
	diskStat *DiskStat,
	partitionStat []*PartitionStat,
	netConnectionsStat []*ConnectionStat,
	processesStat []*ProcessStat) *SystemStat {
	return &SystemStat{
		HostStat:           hostStat,
		UsersStat:          usersStat,
		MemoryStat:         memoryStat,
		CPUStat:            cpuStat,
		DiskStat:           diskStat,
		PartitionStat:      partitionStat,
		NetConnectionsStat: netConnectionsStat,
		ProcessesStat:      processesStat,
	}
}

func (s *SystemStat) ToString() string {
	var users string
	for _, user := range s.UsersStat {
		users += user.ToString() + "\n"
	}

	var partitions string
	for _, partition := range s.PartitionStat {
		partitions += partition.ToString() + "\n"
	}

	var connections string
	for _, connection := range s.NetConnectionsStat {
		connections += connection.ToString() + "\n"
	}

	var processes string
	for _, process := range s.ProcessesStat {
		processes += process.ToString() + "\n"
	}

	return fmt.Sprintf(
		"HostStat: %s\nUsersStat: %s\nMemoryStat: %s\nCPUStat: %s\nDiskStat: %s\nPartitionStat: %s\nNetConnectionsStat: %s\nProcessesStat: %s",
		s.HostStat.ToString(),
		users,
		s.MemoryStat.ToString(),
		s.CPUStat.ToString(),
		s.DiskStat.ToString(),
		partitions,
		connections,
		processes,
	)
}

File: frontend/lib/models/system_stat.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'cpu_stat.dart';
import 'disk_stat.dart';
import 'host_stat.dart';
import 'memory_stat.dart';
import 'net_connection_stat.dart';
import 'partition_stat.dart';
import 'process_stat.dart';
import 'user_stat.dart';

part 'system_stat.freezed.dart';
part 'system_stat.g.dart';

@freezed
class SystemStat with _$SystemStat {
  const factory SystemStat({
    @JsonKey(name: 'host_stat') required HostStat hostStat,
    @JsonKey(name: 'users_stat') required List<UserStat> usersStat,
    @JsonKey(name: 'memory_stat') required MemoryStat memoryStat,
    @JsonKey(name: 'cpu_stat') required CpuStat cpuStat,
    @JsonKey(name: 'disk_stat') required DiskStat diskStat,
    @JsonKey(name: 'partitions_stat')
    required List<PartitionStat> partitionsStat,
    @JsonKey(name: 'net_connections_stat')
    required List<NetConnectionStat> netConnectionsStat,
    @JsonKey(name: 'processes_stat') required List<ProcessStat> processesStat,
  }) = _SystemStat;

  factory SystemStat.fromJson(Map<String, dynamic> json) =>
      _$SystemStatFromJson(json);
}
